homeassistant/components/onvif/event.py

```diff
@@ -226,9 +226,9 @@ async def async_parse_messages(self, messages) -> None:
 
             self._events[event.uid] = event
 
-    def get_uid(self, uid) -> Event:
+    def get_uid(self, uid) -> Event | None:
         """Retrieve event for given id."""
-        return self._events[uid]
+        return self._events.get(uid)
```

Comment:
Not sure why this is needed.  Are there times when we will call `get_uid()` with an invalid uid, and it is more helpful to mask the invalid uid by returning `None`?  Is the exception a problem?

Reply:
When the entity doesn't exist (not sent when we initialize the subscription) it won't yet exist in the Event Manager. The `is_on` and `native_state` methods attempt to retrieve it from the Event Manager, and if it doesn't exist this would throw an exception. Now it returns none and we use the last known state returned from `RestoreEntity` or `RestoreSensor` in `async_added_to_hass`.

Reply:
Is there a way of returning None when we know it's not fully ready? If not this is probably ok 👍

Reply:
That’s kind of what this does since the Events Manager keeps all known events in a dictionary with the unique ID as the key. If it’s not in the dictionary then we haven’t gotten any information about it. If we haven’t gotten any information about it, we should use the restored value. As soon as we receive an ONVIF event, then the Event Manager will add it to the dictionary and the sensor will update with the new state.